### DISCUSSION TITLE
How does this work with the app.app_context

### DISCUSSION THREAD

**Discussion by saromba** [spec-first/connexion#1838] [General]
Hi all,

how does this work with this app.app_context() thing?
The application (swagger ui) is running properly, but when calling an endpoint, it seems that something is not instantiated properly.
I got an error from my utils.py (when calling the database).

```
File "/home/sascha/PycharmProjects/connexion_test/api/utils.py", line 30, in basic_auth
    res = ApiUser.query.filter(ApiUser.id == username).first()
...
RuntimeError: Working outside of application context.

This typically means that you attempted to use functionality that needed
the current application. To solve this, set up an application context
with app.app_context(). See the documentation for more information.
```

Here is my run.py

```
import os
from connexion import FlaskApp
from api import ApiUserManager, ApiUser, db
from auth import auth as auth_object
from dotenv import load_dotenv

load_dotenv()

app = FlaskApp(__name__)

with app.app.app_context():
    app.app.config['SQLALCHEMY_DATABASE_URI'] = os.getenv('database_url')
    app.app.config['USER_EMAIL_SENDER_NAME'] = 'foo'
    app.app.config['USER_EMAIL_SENDER_EMAIL'] = "noreply@example.com"
    app.app.config['SECRET_KEY'] = 'This is an INSECURE secret!! DO NOT use this in production!!'
    user_manager = ApiUserManager(app.app, db, ApiUser, auth_object)
    login_manager = user_manager.login_manager


    @login_manager.user_loader
    def load_user(user_id):
        try:
            return ApiUser.query.filter(ApiUser.id == user_id).first()
        except:
            return None

    db.init_app(app.app)
    db.app = app.app

    app.add_api('my_api.yaml')


app.run()
```

Kind regards,
Sascha

**Comment by ermisp** [comment]
Hello @saromba ,

I had this issue a few time porting a Flask app from 2.14.2 to 3.0.6, it tried the "with app.app.app_context()" but for some reason it did not work in my app.

What did work was one of the answers [here](https://stackoverflow.com/questions/34122949/working-outside-of-application-context-flask) that suggests to add 
`app.app.app_context().push()`
in the code.

In my case putting it anywhere in the run.py worked, hope it helps.

**Comment by cziebuhr** [comment]
I have the same issue and tried it with a custom middleware
```
    class MyMiddleware:
        def __init__(self, next_app):
            self.next_app = next_app

        async def __call__(self, scope, receive, send):
            # TODO exc handling?
            with flask_app.app_context():
                await self.next_app(scope, receive, send)

    app.add_middleware(MyMiddleware, position=MiddlewarePosition.BEFORE_SECURITY)
```
Wouldn't it be better to permanently push the app_context before the other handlers have run? This would also allow the use of the `flask.request` proxy instead of using https://github.com/spec-first/connexion/pull/1883.

**Comment by chrisinmtown** [comment]
For what it's worth, Flask version 3.0 added the requirement of a context. The OP's code refers to many things not shown, like how `db` is created, and does not state whether the app uses library https://pypi.org/project/Flask-SQLAlchemy/    I guess I'm trying to say, it's very difficult to guess at an answer when the information is not complete.

**Comment by chrisinmtown** [comment]
If anyone is still listening/waiting for a response here, you could possibly start with the `examples/sqlalchemy` example, that demonstrates (in a tiny way) how to use Connexion v3, Flask and SQLAlchemy together.
